packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts

```diff
@@ -0,0 +1,74 @@
+import { DAppClient } from '@airgap/beacon-sdk/dist/clients/DappClient';
+import { PermissionResponse } from '@airgap/beacon-sdk/dist/messages/Messages';
+import {
+  createOriginationOperation,
+  createSetDelegateOperation,
+  createTransferOperation,
+  WalletDelegateParams,
+  WalletOriginateParams,
+  WalletProvider,
+  WalletTransferParams,
+} from '@taquito/taquito';
+import { encodeKeyHash } from '@taquito/utils';
+
+export class BeaconWalletNotInitialized implements Error {
+  name = 'BeaconWalletNotInitialized';
+  message = 'You need to initialize BeaconWallet by calling beaconWallet.requestPermissions first';
+}
+
+export class BeaconWallet implements WalletProvider {
+  public client: DAppClient;
+
+  private permissions?: PermissionResponse;
+
+  constructor(name: string) {
+    this.client = new DAppClient(name);
```

Comment:
We need to provide more option to configure the dApp client. Probably allowing the user to pass in the DAppClient instance would be better.

Reply:
Yes, there will probably be more options to configure the dApp client soon (eg. disabling standard behaviour like showing the QR code alert).

What if we introduced a `DAppClientOptions` object that can be passed in the constructor, so you can use the same type in your constructor?

Reply:
Good idea, let's do it that way :+1: